test/control-plane/services/graceful-termination/graceful_termination_test.go

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package graceful_termination
+
+import (
+	"testing"
+
+	"github.com/cilium/cilium/pkg/option"
+	. "github.com/cilium/cilium/test/control-plane/services"
+)
+
+func TestGracefulTermination(t *testing.T) {
+	modConfig := func(c *option.DaemonConfig) {
+		c.EnableK8sTerminatingEndpoint = true
+	}
+	NewGoldenServicesTest(t, "graceful-term-control-plane").Run(t, modConfig)
```

Comment:
A comment saying what each state/golden corresponds to and what the golden test is testing would be useful.. Otherwise, a reader would need to read generate.sh.

Reply:
Done.